pkg/autodiscovery/config_poller.go

```diff
@@ -50,14 +50,62 @@ func (pd *configPoller) stop() {
 }
 
 // start starts polling the provider descriptor
-func (pd *configPoller) start(ac *AutoConfig) {
+func (pd *configPoller) start(ctx context.Context, ac *AutoConfig) {
```

Comment:
nit: rename `pd` to `cp`

```suggestion
func (cp *configPoller) start(ctx context.Context, ac *AutoConfig) {
```